lib/marc_to_argot/macros/unc/call_number.rb

```diff
@@ -0,0 +1,82 @@
+# coding: utf-8
+
+module MarcToArgot
+  module Macros
+    module UNC
+      module CallNumber
+        include MarcToArgot::Util
+
+        # extracts call numbers from an items
+        # and maps them into the output_hash
+        # populates:
+        #   `call_number_schemes`,
+        #   `shelfkey`,
+        #   `reverse_shelfkey`,
+        #   `lcc_callnum_classification`
+
+        def process_call_numbers(rec, cxt)
+          out = cxt.output_hash
+          if out['items']
+            items = out['items'].map{ |i| JSON.parse(i) }
+            item_call_numbers = extract_call_numbers(items)
+          end
+          
+          bib_call_numbers = extract_bib_LC_call_numbers(rec)
+
+
+          unless bib_call_numbers['LC'].empty?
+            if item_call_numbers.nil?
+              item_call_numbers = { 'LC' => bib_call_numbers['LC'] }
+            elsif item_call_numbers.has_key?('LC')
+              bib_call_numbers['LC'].each { |cn| item_call_numbers['LC'] << cn }
+              item_call_numbers['LC'].uniq!
+            else
+              item_call_numbers['LC'] = bib_call_numbers['LC']
+            end
+          end
+
+          if item_call_numbers
+            cxt.output_hash['call_number_schemes'] = item_call_numbers.keys
+            map_shelfkeys!(out, item_call_numbers)
+            map_multi_callnum_classification(out, item_call_numbers)
+          end
+
+        end
+        
+        private
+
+        def extract_bib_LC_call_numbers(rec)
+          lc_call_numbers = []
+          Traject::MarcExtractor.cached('050:090', alternate_script: false).each_matching_line(rec) do |field, spec, extractor|
+            cutter = field['b']
+            cns = field.select{ |sf| sf.code == 'a' }.map{ |sf| sf.value }
+            next if cns.length == 0
+            cns.each do |cn|
+              if cutter
+                lc_call_numbers << "#{cn} #{cutter}"
+              else
+                lc_call_numbers << cn
+              end
+            end
+          end
+          return { 'LC' => lc_call_numbers.map{ |cn| LCC.normalize(cn) }.uniq }
+        end
+
+        def map_multi_callnum_classification(out, call_numbers)
+          return unless call_numbers.key?('LC')
+          raise ValueError, "#{call_numbers} has no LC call numbers" unless call_numbers['LC']

```

Comment:
Do you intend to halt execution here, or do you just want to log this and continue: `logger.warn("#{call_numbers} has no LC call numbers")`?

Reply:
The intention is to stop execution of the rest of the block if there are no LC call numbers. We have no expectation that every bib will contain an LC call number, so I wouldn't want a warning about it.

Reply:
I think `raise ValueError, "#{call_numbers} has no LC call numbers" unless call_numbers['LC']` should be removed. If you want to stop execution of the block if the value of call_numbers['LC'] is nil or empty then I'd change the line above to something like: `return if !call_numbers.key?('LC') || call_numbers['LC'].nil? || call_numbers['LC'].empty?`

Reply:
OH, that! I don't know how, but I was looking at the previous method definition (which I wrote). 

I basically minorly modified the thing you are talking about from (Adam's?) code here: https://github.com/trln/marc-to-argot/blob/master/lib/marc_to_argot/macros/shared/call_numbers.rb#L38

I'll take it out of UNC's because I definitely don't want errors raised.

Reply:
Changes made.